### DISCUSSION TITLE
Link QueryType with new ObjectType

### DISCUSSION THREAD

**Discussion by lasongrzegorz** [mirumee/ariadne#1104] [Q&A]
Hi, having this graphql schema,
```
type Query {
  ping: String!
  ...
  new_query_field: NewQueryField!
}

type NewQueryField {
  total: Int!
  label: String!
}
```
I tried to create a separate QueryType:
```
query = QueryType()
query.set_field("ping", ping)
...
new_query = ObjectType("NewQuery")
new_query.set_field("new_query_data_set_1", resolve_new_query_data_set_1)
new_query.set_field("new_query_data_set_2", resolve_new_query_data_set_2)

ariadne_schema = make_federated_schema(
    type_defs,
    [query, new_query],
)
```
The reason for that is to call for smaller chunks of data within one resolver. This is in opposite to :
```
query = QueryType()
query.set_field("ping", ping)
...
query.set_field("new_query_data_set_1_and_2", resolve_new_query_data_set_1_and_2)

ariadne_schema = make_federated_schema(
    type_defs,
    [query],
)
```

However, I cannot find a way to bind a `new_query` to `query` or make it discoverable in schema? 

Thanks a lot for any advice.

**Comment by rafalp** [comment]
```
query.set_field("new_query_data_set_1_and_2", resolve_new_query_data_set_1_and_2)
```

This is the idiomatic way in GraphQL. Are you having performance problems or running out of memory?

The hack to achieve what you are asking is for, to trigger resolvers for field's types without having data from `query. new_query_data_set_1_and_2 ` is to still set resolver for `Query.new_query_field `, but make this resolver return `True`.

**Reply by lasongrzegorz** [reply]
I am not have any performance problems or running out of memory - it's cleaner for me to have all resolvers assignment/reference in one place.

For the hack, do I get it right, that you propose to simply amend resolvers response as e.g.:

```
def resolve_new_query_data_set_1(...) -> dict:
    return {
        "data_set_1": {"data1": "value1"},
        "data_set_2": None,
    }

def resolve_new_query_data_set_2(...) -> dict:
    return {
        "data_set_1": None,
        "data_set_2": {"data2": "value2"},
    }
```
    
Lastly, just to be sure - there is no nice way to bind/extend existing `query = QueryType()` to/with new query object  `new_query = ObjectType("Query")` ?

**Reply by rafalp** [reply]
> For the hack, do I get it right, that you propose to simply amend resolvers response as e.g.:

No. What I am proposing is adding this to your original code:

```
query.set_field("new_query_field", lambda obj, info: True)
```

This will make the query executor to call your resolvers for `NewQueryField` fields, if they are included in query. Those resolvers will be called with `True` as first argument.

- - - -

> Lastly, just to be sure - there is no nice way to bind/extend existing query = QueryType() to/with new query object new_query = ObjectType("Query") ?

There is:

```graphql
type Query {
  ping: String!
}

extend type Query {
  pong: String!
}
```

```python
query = QueryType()
query.set_field("ping", ping)

new_query = QueryType()
query.set_field("pong", pong)

ariadne_schema = make_federated_schema(
    type_defs,
    [query, new_query],
)
```

This will produce GraphQL schema:

```graphql
type Query {
  ping: String!
  pong: String!
}
```
